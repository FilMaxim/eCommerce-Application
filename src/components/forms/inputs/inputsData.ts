import { minAge } from '../util/validationSchema';

export const inputsData = {
  firstName: {
    name: 'firstName',
    type: 'text',
    placeholder: 'First name'
  },
  lastName: {
    name: 'lastName',
    type: 'text',
    placeholder: 'Last name'
  },
  email: {
    name: 'email',
    type: 'text',
    placeholder: 'Email'
  },
  password: {
    name: 'password',
    type: 'password',
    placeholder: 'Password'
  },
  date: {
    name: 'date',
    type: 'date',
    placeholder: 'Date of Birth'
  },
  addressFieldSet: {
    city: {
      name: 'City',
      type: 'text',
      placeholder: 'City'
    },
    postalCode: {
      name: 'PostalCode',
      type: 'text',
      placeholder: 'Postal Code'
    },
    streetName: {
      name: 'StreetName',
      type: 'text',
      placeholder: 'Street'
    }
  }
};

export enum FieldSetName {
  Shipping = 'shipping',
  Billing = 'billing'
}

export const initialValuesRegistration = {
  firstName: '',
  lastName: '',
  date: minAge,
  email: '',
  password: '',
  shippingStreetName: '',
  shippingCity: '',
  shippingPostalCode: '',
  shippingCountry: '',
  billingStreetName: '',
  billingCity: '',
  billingPostalCode: '',
  billingCountry: '',
  billingStateChecked: true,
  shippingStateChecked: true
};

export enum InputClasses {
  wrapper = 'relative w-[16rem] mb-6',
  error = 'absolute bottom-[-1rem] left-0 text-xs italic text-red-600',
  labelWrapper = 'absolute left-3 top-0 text-xs text-gray-500',
  input = 'leading-tight text-gray-700 transition-all focus:border-secondary focus:outline-none disabled:border-slate-500 disabled:bg-slate-200 w-full appearance-none border-0 border-b-[1px] border-gray-500 px-3  pt-4'
}
