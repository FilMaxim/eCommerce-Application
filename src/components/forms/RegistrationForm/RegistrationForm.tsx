import { Form, Formik } from 'formik';
import { PasswordInput } from '../inputs/PasswordInput';
import { SubmitBtn } from '../inputs/SubmitBtn';
import { AdressFieldSet } from '../inputs/AdressFieldSet';
import { Link } from 'react-router-dom';
import { links } from '../../../utils/links';
import { useState } from 'react';
import { FieldSetName, inputsData } from '../inputs/inputsData';
import { Input } from '../inputs/Input';
import type { RegistrationFormProps } from '../../../utils/types';
import Checkbox from '@mui/material/Checkbox';
import FormControlLabel from '@mui/material/FormControlLabel';

export const RegistrationForm = ({ initialValues, getValidationSchema, onSubmit }: RegistrationFormProps) => {
  const [isSameAddress, setIsSameAddress] = useState(true);
  const handleCheckboxChange = () => {
    setIsSameAddress(!isSameAddress);
  };
  const { firstName, lastName, date, email } = inputsData;

  return (
    <Formik
      initialValues={initialValues}
      validationSchema={getValidationSchema(isSameAddress)}
      onSubmit={onSubmit}
    >
      {(formik) => {
        return (
          <Form className="flex w-96 flex-col justify-center gap-1 rounded-2xl bg-slate-200 px-8 pb-8 pt-6 shadow-[0_2px_15px_-3px_rgba(0,0,0,0.07),0_10px_20px_-2px_rgba(0,0,0,1.1)] max-sm:w-56">
            <Input
              name={firstName.name}
              placeholder={firstName.placeholder}
              type={firstName.type}
              formik={formik}
            />
            <Input
              name={lastName.name}
              placeholder={lastName.placeholder}
              type={lastName.type}
              formik={formik}
            />
            <Input
              name={date.name}
              placeholder={date.placeholder}
              type={date.type}
              formik={formik}
            />
            <Input
              name={email.name}
              placeholder={email.placeholder}
              type={email.type}
              formik={formik}
            />
            <PasswordInput formik={formik} />
            <AdressFieldSet
              fieldSet={FieldSetName.Shipping}
              formik={formik}
            />

            <FormControlLabel
              control={
                <Checkbox
                  name={'toggle'}
                  checked={isSameAddress}
                  onChange={handleCheckboxChange}
                />
              }
              label="same as billing address"
            />
            {!isSameAddress && (
              <AdressFieldSet
                fieldSet={FieldSetName.Billing}
                formik={formik}
              />
            )}
            <SubmitBtn />
            <p className="mt-3 text-center text-[14px]">
              Have an account?
              <Link
                to={links.login.path}
                className="text-link-color"
              >
                {' '}
                Log In
              </Link>
            </p>
          </Form>
        );
      }}
    </Formik>
  );
};
