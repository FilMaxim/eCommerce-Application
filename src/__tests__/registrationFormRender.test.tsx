import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { RegistrationForm } from '../components/forms/RegistrationForm/RegistrationForm';
import { initialValues } from '../components/forms/inputs/inputsData';
import { getValidationSchema } from '../components/forms/util/validationSchema';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

describe('RegistrationForm renders correctly', () => {
  // TODO: fix snapshot
  it.skip('renders correctly', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Routes>
          <Route
            path={''}
            element={
              <RegistrationForm
                initialValues={initialValues}
                onSubmit={jest.fn()}
                getValidationSchema={getValidationSchema}
              />
            }
          />
        </Routes>
      </BrowserRouter>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('expands when billing address checkbox is unchecked', () => {
    render(
      <BrowserRouter>
        <Routes>
          <Route
            path={''}
            element={
              <RegistrationForm
                initialValues={initialValues}
                onSubmit={jest.fn()}
                getValidationSchema={getValidationSchema}
              />
            }
          />
        </Routes>
      </BrowserRouter>
    );

    expect(screen.queryByText(/Billing address:/i)).toBeFalsy();

    fireEvent.click(screen.getByLabelText(/same as billing address/i));

    expect(screen.getByText(/Billing address:/i)).toBeTruthy();
  });
});
