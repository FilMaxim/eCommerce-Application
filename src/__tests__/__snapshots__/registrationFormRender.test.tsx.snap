// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm renders correctly renders correctly 1`] = `
<form
  action="#"
  className="flex w-96 flex-col justify-center gap-1 rounded-2xl bg-slate-200 px-8 pb-8 pt-6 shadow-[0_2px_15px_-3px_rgba(0,0,0,0.07),0_10px_20px_-2px_rgba(0,0,0,1.1)] max-sm:w-56"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <label
    className="text-sm text-gray-500"
    htmlFor="firstName"
  >
    First name
  </label>
  <input
    className="focus:shadow-outline w-full appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow transition-all focus:outline-none"
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="First name"
    type="text"
    value=""
  />
  <label
    className="text-sm text-gray-500"
    htmlFor="lastName"
  >
    Last name
  </label>
  <input
    className="focus:shadow-outline w-full appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow transition-all focus:outline-none"
    name="lastName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Last name"
    type="text"
    value=""
  />
  <label
    className="text-sm text-gray-500"
    htmlFor="date"
  >
    Date of Birth
  </label>
  <input
    className="focus:shadow-outline w-full appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow transition-all focus:outline-none"
    name="date"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Date of Birth"
    type="date"
    value="2005-08-20"
  />
  <label
    className="text-sm text-gray-500"
    htmlFor="email"
  >
    Email
  </label>
  <input
    className="focus:shadow-outline w-full appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow transition-all focus:outline-none"
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <label
    className="text-sm font-bold text-gray-700"
    htmlFor="password"
  >
    Password:
  </label>
  <div
    className="relative"
  >
    <input
      className="focus:shadow-outline w-full appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow focus:outline-none"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      className="absolute right-2 top-2 w-5"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="visibility icon"
        src="no-view.svg"
      />
    </button>
  </div>
  <h3
    className="text-sm font-bold text-gray-700"
  >
    Shipping address:
  </h3>
  <div
    className="mb-[0.125rem] block min-h-[1.5rem] pl-[1.5rem]"
  >
    <label
      className="inline-block pl-[0.15rem] hover:cursor-pointer"
    >
      <input
        checked={true}
        className="checked:border-primary checked:bg-primary dark:checked:border-primary dark:checked:bg-primary relative float-left -ml-[1.5rem] mr-[6px] mt-[0.15rem] h-[1.125rem] w-[1.125rem] appearance-none rounded-[0.25rem] border-[0.125rem] border-solid border-neutral-300 outline-none before:pointer-events-none before:absolute before:h-[0.875rem] before:w-[0.875rem] before:scale-0 before:rounded-full before:bg-transparent before:opacity-0 before:shadow-[0px_0px_0px_13px_transparent] before:content-[''] checked:before:opacity-[0.16] checked:after:absolute checked:after:-mt-px checked:after:ml-[0.25rem] checked:after:block checked:after:h-[0.8125rem] checked:after:w-[0.375rem] checked:after:rotate-45 checked:after:border-[0.125rem] checked:after:border-l-0 checked:after:border-t-0 checked:after:border-solid checked:after:border-white checked:after:bg-transparent checked:after:content-[''] hover:cursor-pointer hover:before:opacity-[0.04] hover:before:shadow-[0px_0px_0px_13px_rgba(0,0,0,0.6)] focus:shadow-none focus:transition-[border-color_0.2s] focus:before:scale-100 focus:before:opacity-[0.12] focus:before:shadow-[0px_0px_0px_13px_rgba(0,0,0,0.6)] focus:before:transition-[box-shadow_0.2s,transform_0.2s] focus:after:absolute focus:after:z-[1] focus:after:block focus:after:h-[0.875rem] focus:after:w-[0.875rem] focus:after:rounded-[0.125rem] focus:after:content-[''] checked:focus:before:scale-100 checked:focus:before:shadow-[0px_0px_0px_13px_#3b71ca] checked:focus:before:transition-[box-shadow_0.2s,transform_0.2s] checked:focus:after:-mt-px checked:focus:after:ml-[0.25rem] checked:focus:after:h-[0.8125rem] checked:focus:after:w-[0.375rem] checked:focus:after:rotate-45 checked:focus:after:rounded-none checked:focus:after:border-[0.125rem] checked:focus:after:border-l-0 checked:focus:after:border-t-0 checked:focus:after:border-solid checked:focus:after:border-white checked:focus:after:bg-transparent dark:border-neutral-600 dark:focus:before:shadow-[0px_0px_0px_13px_rgba(255,255,255,0.4)] dark:checked:focus:before:shadow-[0px_0px_0px_13px_#3b71ca]"
        name="shippingStateChecked"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value={true}
      />
      to save the address as their default address
    </label>
  </div>
  <label
    className="text-sm text-gray-500"
    htmlFor="shippingStreetName"
  >
    Street
  </label>
  <input
    className="focus:shadow-outline w-full appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow transition-all focus:outline-none"
    name="shippingStreetName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Street"
    type="text"
    value=""
  />
  <label
    className="text-sm text-gray-500"
    htmlFor="shippingCity"
  >
    City
  </label>
  <input
    className="focus:shadow-outline w-full appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow transition-all focus:outline-none"
    name="shippingCity"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="City"
    type="text"
    value=""
  />
  <label
    className="text-sm text-gray-500"
    htmlFor="shippingPostalCode"
  >
    Postal Code
  </label>
  <input
    className="focus:shadow-outline w-full appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow transition-all focus:outline-none"
    name="shippingPostalCode"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Postal Code"
    type="text"
    value=""
  />
  <label
    className="text-sm font-bold text-gray-700"
    htmlFor="shippingCountry"
  >
    Country:
  </label>
  <select
    className="focus:shadow-outline appearance-none rounded border border-cyan-500 px-3 py-2 leading-tight text-gray-700 shadow focus:outline-none"
    name="shippingCountry"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Country"
    type="text"
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      Select a country
    </option>
    <option
      value="Afghanistan"
    >
      Afghanistan
    </option>
    <option
      value="Aland Islands"
    >
      Aland Islands
    </option>
    <option
      value="Albania"
    >
      Albania
    </option>
    <option
      value="Algeria"
    >
      Algeria
    </option>
    <option
      value="American Samoa"
    >
      American Samoa
    </option>
    <option
      value="Andorra"
    >
      Andorra
    </option>
    <option
      value="Angola"
    >
      Angola
    </option>
    <option
      value="Anguilla"
    >
      Anguilla
    </option>
    <option
      value="Antarctica"
    >
      Antarctica
    </option>
    <option
      value="Antigua And Barbuda"
    >
      Antigua And Barbuda
    </option>
    <option
      value="Argentina"
    >
      Argentina
    </option>
    <option
      value="Armenia"
    >
      Armenia
    </option>
    <option
      value="Aruba"
    >
      Aruba
    </option>
    <option
      value="Australia"
    >
      Australia
    </option>
    <option
      value="Austria"
    >
      Austria
    </option>
    <option
      value="Azerbaijan"
    >
      Azerbaijan
    </option>
    <option
      value="Bahamas"
    >
      Bahamas
    </option>
    <option
      value="Bahrain"
    >
      Bahrain
    </option>
    <option
      value="Bangladesh"
    >
      Bangladesh
    </option>
    <option
      value="Barbados"
    >
      Barbados
    </option>
    <option
      value="Belarus"
    >
      Belarus
    </option>
    <option
      value="Belgium"
    >
      Belgium
    </option>
    <option
      value="Belize"
    >
      Belize
    </option>
    <option
      value="Benin"
    >
      Benin
    </option>
    <option
      value="Bermuda"
    >
      Bermuda
    </option>
    <option
      value="Bhutan"
    >
      Bhutan
    </option>
    <option
      value="Bolivia"
    >
      Bolivia
    </option>
    <option
      value="Bosnia And Herzegovina"
    >
      Bosnia And Herzegovina
    </option>
    <option
      value="Botswana"
    >
      Botswana
    </option>
    <option
      value="Bouvet Island"
    >
      Bouvet Island
    </option>
    <option
      value="Brazil"
    >
      Brazil
    </option>
    <option
      value="British Indian Ocean Territory"
    >
      British Indian Ocean Territory
    </option>
    <option
      value="Brunei Darussalam"
    >
      Brunei Darussalam
    </option>
    <option
      value="Bulgaria"
    >
      Bulgaria
    </option>
    <option
      value="Burkina Faso"
    >
      Burkina Faso
    </option>
    <option
      value="Burundi"
    >
      Burundi
    </option>
    <option
      value="Cambodia"
    >
      Cambodia
    </option>
    <option
      value="Cameroon"
    >
      Cameroon
    </option>
    <option
      value="Canada"
    >
      Canada
    </option>
    <option
      value="Cape Verde"
    >
      Cape Verde
    </option>
    <option
      value="Cayman Islands"
    >
      Cayman Islands
    </option>
    <option
      value="Central African Republic"
    >
      Central African Republic
    </option>
    <option
      value="Chad"
    >
      Chad
    </option>
    <option
      value="Chile"
    >
      Chile
    </option>
    <option
      value="China"
    >
      China
    </option>
    <option
      value="Christmas Island"
    >
      Christmas Island
    </option>
    <option
      value="Cocos (Keeling) Islands"
    >
      Cocos (Keeling) Islands
    </option>
    <option
      value="Colombia"
    >
      Colombia
    </option>
    <option
      value="Comoros"
    >
      Comoros
    </option>
    <option
      value="Congo"
    >
      Congo
    </option>
    <option
      value="Congo, Democratic Republic"
    >
      Congo, Democratic Republic
    </option>
    <option
      value="Cook Islands"
    >
      Cook Islands
    </option>
    <option
      value="Costa Rica"
    >
      Costa Rica
    </option>
    <option
      value="Cote D'Ivoire"
    >
      Cote D'Ivoire
    </option>
    <option
      value="Croatia"
    >
      Croatia
    </option>
    <option
      value="Cuba"
    >
      Cuba
    </option>
    <option
      value="Cyprus"
    >
      Cyprus
    </option>
    <option
      value="Czech Republic"
    >
      Czech Republic
    </option>
    <option
      value="Denmark"
    >
      Denmark
    </option>
    <option
      value="Djibouti"
    >
      Djibouti
    </option>
    <option
      value="Dominica"
    >
      Dominica
    </option>
    <option
      value="Dominican Republic"
    >
      Dominican Republic
    </option>
    <option
      value="Ecuador"
    >
      Ecuador
    </option>
    <option
      value="Egypt"
    >
      Egypt
    </option>
    <option
      value="El Salvador"
    >
      El Salvador
    </option>
    <option
      value="Equatorial Guinea"
    >
      Equatorial Guinea
    </option>
    <option
      value="Eritrea"
    >
      Eritrea
    </option>
    <option
      value="Estonia"
    >
      Estonia
    </option>
    <option
      value="Ethiopia"
    >
      Ethiopia
    </option>
    <option
      value="Falkland Islands (Malvinas)"
    >
      Falkland Islands (Malvinas)
    </option>
    <option
      value="Faroe Islands"
    >
      Faroe Islands
    </option>
    <option
      value="Fiji"
    >
      Fiji
    </option>
    <option
      value="Finland"
    >
      Finland
    </option>
    <option
      value="France"
    >
      France
    </option>
    <option
      value="French Guiana"
    >
      French Guiana
    </option>
    <option
      value="French Polynesia"
    >
      French Polynesia
    </option>
    <option
      value="French Southern Territories"
    >
      French Southern Territories
    </option>
    <option
      value="Gabon"
    >
      Gabon
    </option>
    <option
      value="Gambia"
    >
      Gambia
    </option>
    <option
      value="Georgia"
    >
      Georgia
    </option>
    <option
      value="Germany"
    >
      Germany
    </option>
    <option
      value="Ghana"
    >
      Ghana
    </option>
    <option
      value="Gibraltar"
    >
      Gibraltar
    </option>
    <option
      value="Greece"
    >
      Greece
    </option>
    <option
      value="Greenland"
    >
      Greenland
    </option>
    <option
      value="Grenada"
    >
      Grenada
    </option>
    <option
      value="Guadeloupe"
    >
      Guadeloupe
    </option>
    <option
      value="Guam"
    >
      Guam
    </option>
    <option
      value="Guatemala"
    >
      Guatemala
    </option>
    <option
      value="Guernsey"
    >
      Guernsey
    </option>
    <option
      value="Guinea"
    >
      Guinea
    </option>
    <option
      value="Guinea-Bissau"
    >
      Guinea-Bissau
    </option>
    <option
      value="Guyana"
    >
      Guyana
    </option>
    <option
      value="Haiti"
    >
      Haiti
    </option>
    <option
      value="Heard Island & Mcdonald Islands"
    >
      Heard Island & Mcdonald Islands
    </option>
    <option
      value="Holy See (Vatican City State)"
    >
      Holy See (Vatican City State)
    </option>
    <option
      value="Honduras"
    >
      Honduras
    </option>
    <option
      value="Hong Kong"
    >
      Hong Kong
    </option>
    <option
      value="Hungary"
    >
      Hungary
    </option>
    <option
      value="Iceland"
    >
      Iceland
    </option>
    <option
      value="India"
    >
      India
    </option>
    <option
      value="Indonesia"
    >
      Indonesia
    </option>
    <option
      value="Iran, Islamic Republic Of"
    >
      Iran, Islamic Republic Of
    </option>
    <option
      value="Iraq"
    >
      Iraq
    </option>
    <option
      value="Ireland"
    >
      Ireland
    </option>
    <option
      value="Isle Of Man"
    >
      Isle Of Man
    </option>
    <option
      value="Israel"
    >
      Israel
    </option>
    <option
      value="Italy"
    >
      Italy
    </option>
    <option
      value="Jamaica"
    >
      Jamaica
    </option>
    <option
      value="Japan"
    >
      Japan
    </option>
    <option
      value="Jersey"
    >
      Jersey
    </option>
    <option
      value="Jordan"
    >
      Jordan
    </option>
    <option
      value="Kazakhstan"
    >
      Kazakhstan
    </option>
    <option
      value="Kenya"
    >
      Kenya
    </option>
    <option
      value="Kiribati"
    >
      Kiribati
    </option>
    <option
      value="Korea"
    >
      Korea
    </option>
    <option
      value="Kuwait"
    >
      Kuwait
    </option>
    <option
      value="Kyrgyzstan"
    >
      Kyrgyzstan
    </option>
    <option
      value="Lao People's Democratic Republic"
    >
      Lao People's Democratic Republic
    </option>
    <option
      value="Latvia"
    >
      Latvia
    </option>
    <option
      value="Lebanon"
    >
      Lebanon
    </option>
    <option
      value="Lesotho"
    >
      Lesotho
    </option>
    <option
      value="Liberia"
    >
      Liberia
    </option>
    <option
      value="Libyan Arab Jamahiriya"
    >
      Libyan Arab Jamahiriya
    </option>
    <option
      value="Liechtenstein"
    >
      Liechtenstein
    </option>
    <option
      value="Lithuania"
    >
      Lithuania
    </option>
    <option
      value="Luxembourg"
    >
      Luxembourg
    </option>
    <option
      value="Macao"
    >
      Macao
    </option>
    <option
      value="Macedonia"
    >
      Macedonia
    </option>
    <option
      value="Madagascar"
    >
      Madagascar
    </option>
    <option
      value="Malawi"
    >
      Malawi
    </option>
    <option
      value="Malaysia"
    >
      Malaysia
    </option>
    <option
      value="Maldives"
    >
      Maldives
    </option>
    <option
      value="Mali"
    >
      Mali
    </option>
    <option
      value="Malta"
    >
      Malta
    </option>
    <option
      value="Marshall Islands"
    >
      Marshall Islands
    </option>
    <option
      value="Martinique"
    >
      Martinique
    </option>
    <option
      value="Mauritania"
    >
      Mauritania
    </option>
    <option
      value="Mauritius"
    >
      Mauritius
    </option>
    <option
      value="Mayotte"
    >
      Mayotte
    </option>
    <option
      value="Mexico"
    >
      Mexico
    </option>
    <option
      value="Micronesia, Federated States Of"
    >
      Micronesia, Federated States Of
    </option>
    <option
      value="Moldova"
    >
      Moldova
    </option>
    <option
      value="Monaco"
    >
      Monaco
    </option>
    <option
      value="Mongolia"
    >
      Mongolia
    </option>
    <option
      value="Montenegro"
    >
      Montenegro
    </option>
    <option
      value="Montserrat"
    >
      Montserrat
    </option>
    <option
      value="Morocco"
    >
      Morocco
    </option>
    <option
      value="Mozambique"
    >
      Mozambique
    </option>
    <option
      value="Myanmar"
    >
      Myanmar
    </option>
    <option
      value="Namibia"
    >
      Namibia
    </option>
    <option
      value="Nauru"
    >
      Nauru
    </option>
    <option
      value="Nepal"
    >
      Nepal
    </option>
    <option
      value="Netherlands"
    >
      Netherlands
    </option>
    <option
      value="Netherlands Antilles"
    >
      Netherlands Antilles
    </option>
    <option
      value="New Caledonia"
    >
      New Caledonia
    </option>
    <option
      value="New Zealand"
    >
      New Zealand
    </option>
    <option
      value="Nicaragua"
    >
      Nicaragua
    </option>
    <option
      value="Niger"
    >
      Niger
    </option>
    <option
      value="Nigeria"
    >
      Nigeria
    </option>
    <option
      value="Niue"
    >
      Niue
    </option>
    <option
      value="Norfolk Island"
    >
      Norfolk Island
    </option>
    <option
      value="Northern Mariana Islands"
    >
      Northern Mariana Islands
    </option>
    <option
      value="Norway"
    >
      Norway
    </option>
    <option
      value="Oman"
    >
      Oman
    </option>
    <option
      value="Pakistan"
    >
      Pakistan
    </option>
    <option
      value="Palau"
    >
      Palau
    </option>
    <option
      value="Palestinian Territory, Occupied"
    >
      Palestinian Territory, Occupied
    </option>
    <option
      value="Panama"
    >
      Panama
    </option>
    <option
      value="Papua New Guinea"
    >
      Papua New Guinea
    </option>
    <option
      value="Paraguay"
    >
      Paraguay
    </option>
    <option
      value="Peru"
    >
      Peru
    </option>
    <option
      value="Philippines"
    >
      Philippines
    </option>
    <option
      value="Pitcairn"
    >
      Pitcairn
    </option>
    <option
      value="Poland"
    >
      Poland
    </option>
    <option
      value="Portugal"
    >
      Portugal
    </option>
    <option
      value="Puerto Rico"
    >
      Puerto Rico
    </option>
    <option
      value="Qatar"
    >
      Qatar
    </option>
    <option
      value="Reunion"
    >
      Reunion
    </option>
    <option
      value="Romania"
    >
      Romania
    </option>
    <option
      value="Russian Federation"
    >
      Russian Federation
    </option>
    <option
      value="Rwanda"
    >
      Rwanda
    </option>
    <option
      value="Saint Barthelemy"
    >
      Saint Barthelemy
    </option>
    <option
      value="Saint Helena"
    >
      Saint Helena
    </option>
    <option
      value="Saint Kitts And Nevis"
    >
      Saint Kitts And Nevis
    </option>
    <option
      value="Saint Lucia"
    >
      Saint Lucia
    </option>
    <option
      value="Saint Martin"
    >
      Saint Martin
    </option>
    <option
      value="Saint Pierre And Miquelon"
    >
      Saint Pierre And Miquelon
    </option>
    <option
      value="Saint Vincent And Grenadines"
    >
      Saint Vincent And Grenadines
    </option>
    <option
      value="Samoa"
    >
      Samoa
    </option>
    <option
      value="San Marino"
    >
      San Marino
    </option>
    <option
      value="Sao Tome And Principe"
    >
      Sao Tome And Principe
    </option>
    <option
      value="Saudi Arabia"
    >
      Saudi Arabia
    </option>
    <option
      value="Senegal"
    >
      Senegal
    </option>
    <option
      value="Serbia"
    >
      Serbia
    </option>
    <option
      value="Seychelles"
    >
      Seychelles
    </option>
    <option
      value="Sierra Leone"
    >
      Sierra Leone
    </option>
    <option
      value="Singapore"
    >
      Singapore
    </option>
    <option
      value="Slovakia"
    >
      Slovakia
    </option>
    <option
      value="Slovenia"
    >
      Slovenia
    </option>
    <option
      value="Solomon Islands"
    >
      Solomon Islands
    </option>
    <option
      value="Somalia"
    >
      Somalia
    </option>
    <option
      value="South Africa"
    >
      South Africa
    </option>
    <option
      value="South Georgia And Sandwich Isl."
    >
      South Georgia And Sandwich Isl.
    </option>
    <option
      value="Spain"
    >
      Spain
    </option>
    <option
      value="Sri Lanka"
    >
      Sri Lanka
    </option>
    <option
      value="Sudan"
    >
      Sudan
    </option>
    <option
      value="Suriname"
    >
      Suriname
    </option>
    <option
      value="Svalbard And Jan Mayen"
    >
      Svalbard And Jan Mayen
    </option>
    <option
      value="Swaziland"
    >
      Swaziland
    </option>
    <option
      value="Sweden"
    >
      Sweden
    </option>
    <option
      value="Switzerland"
    >
      Switzerland
    </option>
    <option
      value="Syrian Arab Republic"
    >
      Syrian Arab Republic
    </option>
    <option
      value="Taiwan"
    >
      Taiwan
    </option>
    <option
      value="Tajikistan"
    >
      Tajikistan
    </option>
    <option
      value="Tanzania"
    >
      Tanzania
    </option>
    <option
      value="Thailand"
    >
      Thailand
    </option>
    <option
      value="Timor-Leste"
    >
      Timor-Leste
    </option>
    <option
      value="Togo"
    >
      Togo
    </option>
    <option
      value="Tokelau"
    >
      Tokelau
    </option>
    <option
      value="Tonga"
    >
      Tonga
    </option>
    <option
      value="Trinidad And Tobago"
    >
      Trinidad And Tobago
    </option>
    <option
      value="Tunisia"
    >
      Tunisia
    </option>
    <option
      value="Turkey"
    >
      Turkey
    </option>
    <option
      value="Turkmenistan"
    >
      Turkmenistan
    </option>
    <option
      value="Turks And Caicos Islands"
    >
      Turks And Caicos Islands
    </option>
    <option
      value="Tuvalu"
    >
      Tuvalu
    </option>
    <option
      value="Uganda"
    >
      Uganda
    </option>
    <option
      value="Ukraine"
    >
      Ukraine
    </option>
    <option
      value="United Arab Emirates"
    >
      United Arab Emirates
    </option>
    <option
      value="United Kingdom"
    >
      United Kingdom
    </option>
    <option
      value="United States"
    >
      United States
    </option>
    <option
      value="United States Outlying Islands"
    >
      United States Outlying Islands
    </option>
    <option
      value="Uruguay"
    >
      Uruguay
    </option>
    <option
      value="Uzbekistan"
    >
      Uzbekistan
    </option>
    <option
      value="Vanuatu"
    >
      Vanuatu
    </option>
    <option
      value="Venezuela"
    >
      Venezuela
    </option>
    <option
      value="Viet Nam"
    >
      Viet Nam
    </option>
    <option
      value="Virgin Islands, British"
    >
      Virgin Islands, British
    </option>
    <option
      value="Virgin Islands, U.S."
    >
      Virgin Islands, U.S.
    </option>
    <option
      value="Wallis And Futuna"
    >
      Wallis And Futuna
    </option>
    <option
      value="Western Sahara"
    >
      Western Sahara
    </option>
    <option
      value="Yemen"
    >
      Yemen
    </option>
    <option
      value="Zambia"
    >
      Zambia
    </option>
    <option
      value="Zimbabwe"
    >
      Zimbabwe
    </option>
  </select>
  <div
    className="mb-[0.125rem] block min-h-[1.5rem] pl-[1.5rem]"
  >
    <label
      className="inline-block pl-[0.15rem] hover:cursor-pointer"
    >
      <input
        checked={true}
        className="checked:border-primary checked:bg-primary dark:checked:border-primary dark:checked:bg-primary relative float-left -ml-[1.5rem] mr-[6px] mt-[0.15rem] h-[1.125rem] w-[1.125rem] appearance-none rounded-[0.25rem] border-[0.125rem] border-solid border-neutral-300 outline-none before:pointer-events-none before:absolute before:h-[0.875rem] before:w-[0.875rem] before:scale-0 before:rounded-full before:bg-transparent before:opacity-0 before:shadow-[0px_0px_0px_13px_transparent] before:content-[''] checked:before:opacity-[0.16] checked:after:absolute checked:after:-mt-px checked:after:ml-[0.25rem] checked:after:block checked:after:h-[0.8125rem] checked:after:w-[0.375rem] checked:after:rotate-45 checked:after:border-[0.125rem] checked:after:border-l-0 checked:after:border-t-0 checked:after:border-solid checked:after:border-white checked:after:bg-transparent checked:after:content-[''] hover:cursor-pointer hover:before:opacity-[0.04] hover:before:shadow-[0px_0px_0px_13px_rgba(0,0,0,0.6)] focus:shadow-none focus:transition-[border-color_0.2s] focus:before:scale-100 focus:before:opacity-[0.12] focus:before:shadow-[0px_0px_0px_13px_rgba(0,0,0,0.6)] focus:before:transition-[box-shadow_0.2s,transform_0.2s] focus:after:absolute focus:after:z-[1] focus:after:block focus:after:h-[0.875rem] focus:after:w-[0.875rem] focus:after:rounded-[0.125rem] focus:after:content-[''] checked:focus:before:scale-100 checked:focus:before:shadow-[0px_0px_0px_13px_#3b71ca] checked:focus:before:transition-[box-shadow_0.2s,transform_0.2s] checked:focus:after:-mt-px checked:focus:after:ml-[0.25rem] checked:focus:after:h-[0.8125rem] checked:focus:after:w-[0.375rem] checked:focus:after:rotate-45 checked:focus:after:rounded-none checked:focus:after:border-[0.125rem] checked:focus:after:border-l-0 checked:focus:after:border-t-0 checked:focus:after:border-solid checked:focus:after:border-white checked:focus:after:bg-transparent dark:border-neutral-600 dark:focus:before:shadow-[0px_0px_0px_13px_rgba(255,255,255,0.4)] dark:checked:focus:before:shadow-[0px_0px_0px_13px_#3b71ca]"
        name="toggle"
        onChange={[Function]}
        type="checkbox"
        value="false"
      />
      to use the same address for both billing and shipping
    </label>
  </div>
  <button
    className="mt-3 w-full self-center rounded bg-blue-500 px-4 py-2 font-bold text-white transition-all hover:bg-blue-700"
    type="submit"
  >
    Submit
  </button>
  <p
    className="mt-3 text-center text-[14px]"
  >
    Have an account?
    <a
      className="text-link-color"
      href="/login"
      onClick={[Function]}
    >
       
      Log In
    </a>
  </p>
</form>
`;
